src/praisonai-sandbox/praisonai_sandbox/docker.py

```
@@ -216,9 +218,21 @@ async def execute(
             except asyncio.TimeoutError:
+                # Kill the container, not just the client. proc.kill() ends the
+                # docker CLI while the container it started keeps running,
+                # unnamed and unreachable.
+                try:
+                    killer = await asyncio.create_subprocess_exec(
+                        "docker", "kill", container_name,
+                        stdout=asyncio.subprocess.DEVNULL,
+                        stderr=asyncio.subprocess.DEVNULL,
+                    )
+                    await killer.wait()
+                except Exception:  # pragma: no cover - best effort teardown
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)